programs/test/header_test.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ *  A C++ program that includes all of the mbed TLS header files, in order to
+ *  test if no errors are raised in the process.
+ *
```

Comment:
Rather than say we want to check we build without errors...
~~~
/*
 *  A C++ program that includes all of the mbed TLS header files, in order to
 *  test if no errors are raised in the process.
~~~
I'd like to state the purpose more clearly, so I'd propose:
~~~
/*
 * This program is a dummy C++ program to ensure Mbed TLS library header files
 * can be included and built with a C++ compiler.
~~~

Reply:
Done, updated as suggested.